source/application/AVIDumping/AVIDumper.cpp

```diff
@@ -755,531 +829,570 @@ static CRITICAL_SECTION s_fqvCS;
-			Config::localTASflags.aviMode &= ~2;
-			tasFlagsDirty = true;
-			return -1;
-		}
-
-		hr = AVIStreamSetFormat(aviSoundStream, 0, outputFormat, outputFormatSize);
-		if (FAILED(hr))
-		{ 
+            NormalMessageBox(
+                L"AVIFileCreateStream(audio) failed!\nCapture will continue without audio\n",
+                L"Error",
+                MB_OK | MB_ICONERROR);
+            Config::localTASflags.aviMode &= ~2;
+            tasFlagsDirty = true;
+            return -1;
+        }
+
+        hr = AVIStreamSetFormat(aviSoundStream, 0, outputFormat, outputFormatSize);
+        if (FAILED(hr))
+        {
             DebugLog() << "AVIStreamSetFormat(audio) failed!";
-			NormalMessageBox(L"AVIStreamSetFormat(audio) failed!\nCapture will continue without audio\n", L"Error", MB_OK|MB_ICONERROR);
-			Config::localTASflags.aviMode &= ~2;
-			tasFlagsDirty = true;
-			return -1;
-		}
-	}
-
-	aviSoundSampleCount = 0;
-	aviSoundSecondsCount = 0;
-	aviSoundFrameCount = 0;
-
-	return 1;
+            NormalMessageBox(
+                L"AVIStreamSetFormat(audio) failed!\nCapture will continue without audio\n",
+                L"Error",
+                MB_OK | MB_ICONERROR);
+            Config::localTASflags.aviMode &= ~2;
+            tasFlagsDirty = true;
+            return -1;
+        }
+    }
+
+    aviSoundSampleCount = 0;
+    aviSoundSecondsCount = 0;
+    aviSoundFrameCount = 0;
+
+    return 1;
 }
 
 void WriteAVIAudio()
 {
-	if (!aviSoundStream)
-	{
-		int res = OpenAVIAudioStream();
-		if (!res)
-		{
-			return;
-		}
-
-		if (res < 0)
-		{
-			Config::localTASflags.aviMode &= ~2;
-			tasFlagsDirty = true;
-			return;
-		}
-	}
-
-	if (aviCompressedStream && (aviSoundFrameCount < 30 || aviSoundFrameCount+8 < aviFrameCount) && !aviSplitCount)
-	{
-		if (aviSoundFrameCount < aviFrameCount)
-		{
-			AutoCritSect cs(&s_fqaCS); // critical section and check again in case we got here while CloseAVI is running
-			if (aviCompressedStream && (aviSoundFrameCount < 30 || aviSoundFrameCount + 8 < aviFrameCount) && !aviSplitCount)
-			{
-				while (aviSoundFrameCount < aviFrameCount)
-				{
-					// in case video started before audio
-					aviFrameQueue->FillEmptyAudioFrame();
-				}
-			}
-		}
-	}
-
-	aviFrameQueue->FillAudioFrame();
+    if (!aviSoundStream)
+    {
+        int res = OpenAVIAudioStream();
+        if (!res)
+        {
+            return;
+        }
+
+        if (res < 0)
+        {
+            Config::localTASflags.aviMode &= ~2;
+            tasFlagsDirty = true;
+            return;
+        }
+    }
+
+    if (aviCompressedStream && (aviSoundFrameCount < 30 || aviSoundFrameCount + 8 < aviFrameCount)
+        && !aviSplitCount)
+    {
+        if (aviSoundFrameCount < aviFrameCount)
+        {
+            AutoCritSect cs(&s_fqaCS); // critical section and check again in case we got here while
```

Comment:
See if we can prioritize moving comment to an empty line above before breaking it.

Reply:
Doesn't seem to be an option for moving comments to an empty line.